feat: OPTIC-1217: Implement Color design tokens as CSS variables and Dark Mode

yyassi-heartex opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend



### Describe the reason for change
significant improvement to ui tokens + adding the ability to toggle between Auto, Light and Dark mode

robot-ci-heartex commented:
Follow Merge downstream workflow has been failed.
> [Workflow run](https://github.com/HumanSignal/label-studio-enterprise/actions/runs/11481678479)

yyassi-heartex commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/11482033249)
> Successfully merged:  7 files changed, 13 insertions(+), 13 deletions(-)

robot-ci-heartex commented:
Follow Merge downstream workflow has been failed.
> [Workflow run](https://github.com/HumanSignal/label-studio-enterprise/actions/runs/11482086685)

robot-ci-heartex commented:
Follow Merge downstream workflow has been failed.
> [Workflow run](https://github.com/HumanSignal/label-studio-enterprise/actions/runs/11487323348)

robot-ci-heartex commented:
Follow Merge downstream workflow has been failed.
> [Workflow run](https://github.com/HumanSignal/label-studio-enterprise/actions/runs/11487851251)

yyassi-heartex commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/11504540702)
> Successfully merged:  create mode 100644 web/apps/labelstudio/src/pages/DataManager/api-config.example.js

ricardoantoniocm committed:
Adds dark mode support to pages and components.

ricardoantoniocm committed:
Refactors folder-empty icon.

ricardoantoniocm committed:
Adds descriptions to Semantic tokens.

ricardoantoniocm committed:
Adapts elements to dark mode.

ricardoantoniocm committed:
Adapts elements to dark mode.

ricardoantoniocm committed:
Adapts elements to dark mode.

ricardoantoniocm committed:
Adapts elements to dark mode

yyassi-heartex committed:
ThemeToggle can now tell others what theme is currently in use

yyassi-heartex committed:
adding color definition for darkvs light mode

ricardoantoniocm committed:
Adapts elements to dark mode.

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1217/implement-design-tokens-as-css-variables

robot-ci-heartex commented:
Follow Merge downstream workflow has been failed.
> [Workflow run](https://github.com/HumanSignal/label-studio-enterprise/actions/runs/12715996693)

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
lint cleanup

ricardoantoniocm committed:
Adds missing closing element.

ricardoantoniocm committed:
Adds maximum width to User Account page.

ricardoantoniocm committed:
Adapts elements to dark mode.

ricardoantoniocm committed:
Adapts elements to Dark Mode.

ricardoantoniocm committed:
Improves language of completed state for labeling queue.

ricardoantoniocm committed:
Lint fixes.

ricardoantoniocm committed:
Removes empty line.

ricardoantoniocm committed:
lint cleanup

ricardoantoniocm committed:
Adds explicit button type for themeToggle

ricardoantoniocm committed:
Adds explicit button type for MultiStateToggle

ricardoantoniocm committed:
Lint Cleanup

ricardoantoniocm commented:
/deploy
> '/deploy' is an unknown pull-request command.
> See '/help'

ricardoantoniocm commented:
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/12813886539)
> Docker image was pushed with the tag `20250116.170512-fb-optic-1217-implement-d-5ebbcf275`

ricardoantoniocm committed:
Adds dark mode support to multiple components.

yyassi-heartex committed:
adding new css vars for ThemeToggle

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1217/implement-design-tokens-as-css-variables

yyassi-heartex committed:
default theme is now Light

yyassi-heartex committed:
default theme is light

ricardoantoniocm committed:
Adds dark mode support to multiple components.

ricardoantoniocm committed:
Adds dark mode support to multiple components.

ricardoantoniocm committed:
Adds dark mode support to multiple components.

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
remove duplicate selectors

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
conflict updates

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
fix resizer panel border colors and size on hover

bmartel committed:
align border colors for panels and resizer element

bmartel committed:
improve left side panel snap indicator getting stuck when clicking on a tab

bmartel committed:
fix the tab panel group vertical resizer hover border to match the sidepanel horizontal resizer

bmartel committed:
remove hardcoded styles in DM action buttons

bmartel committed:
fix annotation button menu trigger border

bmartel committed:
fix bottombar spacing, remove hardcoded tokens

bmartel committed:
fix tooltip and contextmenu

bmartel committed:
linting

bmartel committed:
fix editor button borders, and the treeview region index text color

bmartel committed:
removing the theme color option from Tooltip scss, leave it to color scheme

bmartel committed:
fix logo from merge

bmartel committed:
fix DM Resizer spacing and positioning

bmartel committed:
fix coloring of headings getting overridden by antd, also toolbar cleanup

bmartel committed:
create project visual config select and input colors

bmartel committed:
generating raw values for surface collection color tokens

bmartel committed:
replacing incorrect design tokens

bmartel committed:
replacing incorrect design tokens

bmartel committed:
replacing incorrect design tokens

bmartel committed:
replacing incorrect design tokens

ricardoantoniocm committed:
Fixes app background. Fixes scrollbars.

bmartel committed:
DM button and filters cleanup

bmartel committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of github.com:HumanSignal/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
themetoggle font size, form select style

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
make homepage components dark mode compatible

bmartel committed:
fix button outline styles

bmartel committed:
fix button outline styles

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
fix InputFile to align with Dark Mode styles

bmartel committed:
fix Counter styles to be dark mode compatible

bmartel committed:
fix icon collapse/expand color

bmartel committed:
fix all projects icon

bmartel committed:
remove static main.css and uikit.css from base template

bmartel committed:
move codemirror to ui lib so it can be themed and shared across the codebase

bmartel committed:
fix code editor base styles to ensure it works in all the use cases we have

bmartel committed:
ensure forwardref is handled

bmartel committed:
allow accent colors to expose raw variables for translucent colors

ricardoantoniocm committed:
Reduces left padding on Home page quick action buttons.

bmartel committed:
fix CreateProject fields

bmartel committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of github.com:HumanSignal/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
fix link styles

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
fix input file styles

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Replaced hard coded color for currentColor for chevron-left-bold and chevron-right-bold icons.

bmartel committed:
add color utility to calculate an accessible color from a bg and fg color set

bmartel committed:
add bg colors as raw variants

bmartel committed:
removing color util, it wont quite work out

bmartel committed:
commentform and textarea focus styles, region editor and detail panel colors

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Improves main navigation icons' contrast.

ricardoantoniocm committed:
Improves display of selected tasks on grid view.

ricardoantoniocm committed:
Improves copy/duplicate and thumbs up/down icons.

ricardoantoniocm committed:
Fixes padding of help icon on table headers.

ricardoantoniocm committed:
Sets Bulk Labeling icon to currentColor.

ricardoantoniocm committed:
Adapts default Tag colors to dark mode.

ricardoantoniocm committed:
Adapts Filters and Select components to dark mode.

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
fix base button padding, adjust only the homepage action buttons instead

bmartel committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of github.com:HumanSignal/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
adding more icons that were found in static folder

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
border colors

bmartel committed:
toast color updates, adding to storybook

bmartel committed:
update toast stories to be under UI namespace

bmartel committed:
adding new semantic tokens

ricardoantoniocm committed:
Adds temporary shadow-intensity token.

ricardoantoniocm committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of https://github.com/heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
wrong styles for CodeBlock

bmartel committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of github.com:HumanSignal/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
linting

ricardoantoniocm committed:
Adds shadow-intensity to shadows to handle different intensity depending on the theme.

ricardoantoniocm committed:
Adapts divider on Membership Info page to dark mode.

ricardoantoniocm committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of https://github.com/heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
fix some of the code-editor styles

bmartel committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of github.com:HumanSignal/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Updates design tokens to improve readability of text in dark mode.

ricardoantoniocm committed:
Adds color-neutral-on-dark-icon to support icons that need to stay white.

ricardoantoniocm committed:
Improves contrast of icons on dark mode.

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Fixes colors in annotation history.

ricardoantoniocm committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of https://github.com/heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Adapts 404 error page to dark mode.

ricardoantoniocm committed:
Improves contrast of text on dark mode.

ricardoantoniocm committed:
Fixes background of application.

ricardoantoniocm committed:
Improves look of links on descriptions.

AndrejOros commented:
/fm sync 
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14493447299)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14493447299

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
fix Homepage action buttons

ricardoantoniocm commented:
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14494154922)
> Docker image was pushed with the tag `20250416.134017-fb-optic-1217-implement-d-1820cb5f1`

bmartel committed:
missing border on code-editor

bmartel committed:
fix App Pagination border color

ricardoantoniocm committed:
Improves text color of modal footers.

ricardoantoniocm committed:
Improves tour styles.

ricardoantoniocm committed:
Adapts react-datepicker on DM to dark mode.

ricardoantoniocm committed:
Replaces primitives with semantic tokens.

ricardoantoniocm committed:
Improves contrast of colors in dark mode.

ricardoantoniocm committed:
Linting.

ricardoantoniocm committed:
Adapts video timeline to dark mode.

yyassi-heartex committed:
minor tweak for selected user height

yyassi-heartex committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of github.com:heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Fixes error breaking shadows on DM dropdowns.

ricardoantoniocm committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of https://github.com/heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

yyassi-heartex committed:
minor tweaks for filter padding

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14510136788)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1217/implement-design-tokens-as-css-variables'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14510136788

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14510136788

ricardoantoniocm committed:
Fixes text color on editor.

ricardoantoniocm committed:
Fixes antd overrides for "result-title".

bmartel committed:
putting ThemeToggle behind a Feature Flag

ricardoantoniocm committed:
Reduces contrast of color-neutral-surface on dark mode.

ricardoantoniocm committed:
Adds missing line at EOF.

ricardoantoniocm committed:
Adds color-accent-gradient.

ricardoantoniocm committed:
Adapts joyride to dark mode.

ricardoantoniocm committed:
Fixes color-accent-gradient token.

ricardoantoniocm committed:
Removes unnecesary comma.

ricardoantoniocm committed:
Fixes overlay not displaying. Improves overall style of the tour steps.

ricardoantoniocm committed:
Improves create prompt tour.

ricardoantoniocm committed:
Adapts textarea to dark mode.

ricardoantoniocm committed:
Fixes Grid Preview button hover state.

ricardoantoniocm committed:
Adds hover state to grid view cells.

AndrejOros commented:
/fm sync

ricardoantoniocm committed:
Fixes border shown on DM data header columns on hover.

bmartel committed:
fix ff imports

ricardoantoniocm committed:
Removes duplicitous styles.

ricardoantoniocm committed:
Adds tokens to support gradients.

ricardoantoniocm committed:
Adapts style of Enhance Prompt icon to support Dark Mode.

ricardoantoniocm committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of https://github.com/heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Adds missing hover effect to buttons on DM.

ricardoantoniocm committed:
Improves text color of negative menu items.

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Fixes prev and next buttons not being visible on dark mode.

ricardoantoniocm committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of https://github.com/heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Fixes file input taking more space than it needs.

ricardoantoniocm committed:
Improves style of destructive buttons.

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1217/implement-design-tokens-as-css-variables

yyassi-heartex committed:
merge conflict resolutions

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
fixing -spacing class name

bmartel committed:
fixing spacing variables and linting

bmartel committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of github.com:HumanSignal/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

yyassi-heartex committed:
adding testid for resources card

yyassi-heartex committed:
making sure valueref is updated

yyassi-heartex committed:
missed this type

ricardoantoniocm committed:
Fixes border not being displayed for Skip, Reject buttons and improves hover states.

ricardoantoniocm committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of https://github.com/heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

bmartel committed:
fix danger button border

ricardoantoniocm committed:
Adapts Link State component to Dark Mode.

ricardoantoniocm committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of https://github.com/heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Fixes hover state of previous and next buttons on View All screen.

yyassi-heartex committed:
select now supports select first option if no value provided

yyassi-heartex committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of github.com:heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Fixes checkbox labels not being visible on dark mode.

ricardoantoniocm committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of https://github.com/heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Ads missing EOF line.

ricardoantoniocm committed:
Fixes rectangle tool, 3 point rectangle tool, rectangle tool smart and 3 point rectangle tool smart icons, default, active and hover states. Fixes zoom active state.

ricardoantoniocm committed:
Adds border to code-editor component.

bmartel committed:
fixing native date color scheme

bmartel committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of github.com:HumanSignal/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

yyassi-heartex committed:
visual tweak

yyassi-heartex committed:
addressing create project button padding

yyassi-heartex committed:
slight visual tweaks

yyassi-heartex committed:
button madneess LSO

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
cleaning up icon size for annotationButton

yyassi-heartex commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14637192838)
> Successfully merged:  create mode 100644 docs/themes/v2/source/images/releases/2-23-wif.png

yyassi-heartex committed:
Merge branch 'develop' into 'fb-optic-1217/implement-design-tokens-as-css-variables'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14637192838

yyassi-heartex committed:
making test more stable

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

yyassi-heartex committed:
cleanup test

ricardoantoniocm committed:
Adds label color to index container.

ricardoantoniocm committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of https://github.com/heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Makes readonly fields look disabled.

ricardoantoniocm committed:
Makes readonly fields look disabled.

ricardoantoniocm committed:
Refactors border for avatars.

ricardoantoniocm committed:
Fixes shadow of volume modal on audio player.

ricardoantoniocm committed:
Adapts colors of audio player when loading to Dark mode.

ricardoantoniocm committed:
Fixes style of View All button.

ricardoantoniocm committed:
Fixes color of links on input labels.

ricardoantoniocm committed:
Fixes hover color of input label links.

ricardoantoniocm committed:
Adds missing bottom padding to input labels on project settings.

ricardoantoniocm committed:
Fixes hover effect of relation item.

ricardoantoniocm committed:
Adapts video settings to Dark mode.

ricardoantoniocm committed:
Fixes disabled state for toggle.

yyassi-heartex committed:
config updated input

yyassi-heartex committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of github.com:heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

yyassi-heartex committed:
export modal cleanup

yyassi-heartex committed:
tweaking colors

yyassi-heartex committed:
slight color tweak for alert error button hover

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14659153456)

AndrejOros committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14659153456

AndrejOros committed:
Merge branch 'develop' into 'fb-optic-1217/implement-design-tokens-as-css-variables'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14659153456

ricardoantoniocm committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of https://github.com/heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Adapts Number input control tag to dark mode.

ricardoantoniocm committed:
Adapts Datetime component to dark mode.

ricardoantoniocm committed:
Adapts number input controls to dark mode.

ricardoantoniocm committed:
Adapts Code Editor code completion popover to dark mode.

ricardoantoniocm committed:
Improves code editor code completion hints style.

bmartel committed:
ensure the head tag script for setting the color scheme is behind the theme toggle feature flag

yyassi-heartex committed:
audio darkmode colors

yyassi-heartex committed:
switching tokens

yyassi-heartex committed:
Merge branch 'fb-optic-1217/implement-design-tokens-as-css-variables' of github.com:heartexlabs/label-studio into fb-optic-1217/implement-design-tokens-as-css-variables

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14702246262)

AndrejOros committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14702246262

ricardoantoniocm committed:
Adds Beta tag to theme toggle.

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1217/implement-design-tokens-as-css-variables

ricardoantoniocm committed:
Fixes Update button not looking properly disabled.

ricardoantoniocm committed:
Improves look and feel of annotation tabs.

yyassi-heartex committed:
tweaking spinner body background

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
updating colors for form validation

yyassi-heartex committed:
slight tweak to the integration test to make it more consistent

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1217/implement-design-tokens-as-css-variables

yyassi-heartex committed:
disabling flakey test again

ricardoantoniocm committed:
Fixes annotation button contents from moving when switching from default to selected state. Improves background of the TopBar.
